File: src/cli/tests/fixtures/doc/csharp/doc006_placeholders.cs
// DOC006 for C#: placeholder markers in doc comments read as finished API
// documentation.
namespace Fixtures;

/// <summary>Holds task methods.</summary>
public class Tasks
{
    /// <summary>TODO: describe this.</summary>
    public void Todo() { }

    /// <summary>FIXME: describe this.</summary>
    public void Fixme() { }

    /// <summary>TBD: describe this.</summary>
    public void Tbd() { }

    /// <summary>Done: described.</summary>
    public void Done() { }
}

File: src/cli/tests/fixtures/doc/rust/doc006_placeholders.rs
//! Rule: DOC006 - doc comments must not contain placeholder text.
//!
//! Placeholder markers (`TODO`, `FIXME`, `TBD`) in doc comments ship
//! incomplete documentation that reads as finished. Items whose doc comments
//! contain a placeholder are flagged.
//!
//! Expected diagnostics:
//! - DOC006 on `pub fn todo_task` (TODO in doc)
//! - DOC006 on `pub fn fixme_task` (FIXME in doc)
//! - DOC006 on `pub fn tbd_task` (TBD in doc)
//!
//! Not flagged (should pass):
//! - `pub fn done` (clean doc)
//! - `pub struct Placeholder` (literal `...` is idiomatic, not a marker)

/// TODO: implement this.
pub fn todo_task() {}

/// FIXME: this is broken.
pub fn fixme_task() {}

/// TBD: what should this do?
pub fn tbd_task() {}

/// Something ... to fill in later.
pub struct Placeholder {
    x: i32,
}

/// A complete function with no placeholders.
pub fn done() {}
